PROTO_tests/tests/numeric_test.py

```diff
@@ -197,22 +204,249 @@ def test_cumprod(self, num_type):
         with pytest.raises(TypeError):
             ak.cos(np.array([range(0, 10)]).astype(num_type))
 
+    @pytest.mark.parametrize("num_type", NO_BOOL)
+    def test_tan(self, num_type):
+        na = NP_TRIG_ARRAYS[num_type]
+        pda = ak.array(na, dtype=num_type)
+
+        assert np.allclose(np.tan(na), ak.tan(pda).to_ndarray(), equal_nan=True)
+        with pytest.raises(TypeError):
+            ak.tan(np.array([range(0, 10)]).astype(num_type))
+
+    @pytest.mark.parametrize("num_type", NO_BOOL)
+    def test_arcsin(self, num_type):
+        if num_type == ak.uint64:
+            na = np.arange(0, 2).astype(num_type)
+        elif num_type == ak.float64:
+            na = np.linspace(-1, 1)
+        else:
+            na = np.arange(-1, 2).astype(num_type)
+        pda = ak.array(na, dtype=num_type)
+
+        assert np.allclose(np.arcsin(na), ak.arcsin(pda).to_ndarray(), equal_nan=True)
+        with pytest.raises(TypeError):
+            ak.arcsin(np.array([range(0, 10)]).astype(num_type))
+
+        # Edge case: Infinities and NaNs
+        if num_type == ak.float64:
+            na = np.array([np.inf, np.nan, -np.inf])
+            pda = ak.array(na)
+            assert np.allclose(np.arcsin(na), ak.arcsin(pda).to_ndarray(), equal_nan=True)
+
+    @pytest.mark.parametrize("num_type", NO_BOOL)
+    def test_arccos(self, num_type):
+        if num_type == ak.uint64:
+            na = np.arange(0, 2).astype(num_type)
+        elif num_type == ak.float64:
+            na = np.linspace(-1, 1)
+        else:
+            na = np.arange(-1, 2).astype(num_type)
+        pda = ak.array(na, dtype=num_type)
+
+        assert np.allclose(np.arccos(na), ak.arccos(pda).to_ndarray(), equal_nan=True)
+        with pytest.raises(TypeError):
+            ak.arccos(np.array([range(0, 10)]).astype(num_type))
+
+        # Edge case: Infinities and NaNs
+        if num_type == ak.float64:
+            na = np.array([np.inf, np.nan, -np.inf])
+            pda = ak.array(na)
+            assert np.allclose(np.arccos(na), ak.arccos(pda).to_ndarray(), equal_nan=True)
+
+    @pytest.mark.parametrize("num_type", NO_BOOL)
+    def test_arctan(self, num_type):
+        na = NP_TRIG_ARRAYS[num_type]
+        pda = ak.array(na, dtype=num_type)
+
+        assert np.allclose(np.arctan(na), ak.arctan(pda).to_ndarray(), equal_nan=True)
+        with pytest.raises(TypeError):
+            ak.arctan(np.array([range(0, 10)]).astype(num_type))
+
+        # Edge case: infinities and NaNs
+        if num_type == ak.float64:
+            na = np.array([np.inf, np.nan, -np.inf])
+            pda = ak.array(na)
+            assert np.allclose(np.arctan(na), ak.arctan(pda).to_ndarray(), equal_nan=True)
+
+    @pytest.mark.parametrize("num_type", NO_BOOL)
+    @pytest.mark.parametrize("denom_type", NO_BOOL)
+    def test_arctan2(self, num_type, denom_type):
+        na_num = NP_TRIG_ARRAYS[num_type]
+        na_denom = np.arange(0, 10).astype(denom_type)
```

Comment:
I don't love hardcoding the `na_denom` to be fixed to be fixed 0..10 of whatever type. Especially because it means `denom[0]` below ill only ever test the scalar case when it is equal to 0. I think it would be better to take random permutations of the representative arrays, so something the below instead
```python
na_num = np.random.permutation(NP_TRIG_ARRAYS[num_type])
na_denom = np.random.permutation(NP_TRIG_ARRAYS[denom_type])
```